app/javascript/components/schedule-form/helper.js

```diff
@@ -0,0 +1,252 @@
+import { API, http } from '../../http_api';
+
+/** Function which returns the data needed for timezone drop-down-list */
+export const timeZoneData = (timezones) => timezones.map((timezone) => ({ value: timezone.name, label: timezone.description }));
+
+/** Function to check if an item is array. if yes, returns true, else returns false */
+const isArray = (item) => Array.isArray(item);
+
+/** To swap the index of the recieved array when api gives the array ['value', 'label]
+* occurs when we select the sub action options */
+const swapIndex = (valueFirst) => {
+  let indexes = { label: 0, value: 1 };
+  if (valueFirst) {
+    indexes = { label: 1, value: 0 };
+  }
+  return indexes;
+};
+
+/** Function to get the options.
+ * The data is either array of strings eg: ['aaa','bbb'] or
+ * array of string arrays eg: [['aaa','bbb], ['ccc','ddd']] */
+export const restructureOptions = (data, valueFirst = false) => {
+  const index = swapIndex(valueFirst);
+  if (!data) { return []; } return data.map((item) => (
+    isArray(item)
+      ? ({ label: item[index.label], value: item[index.value] })
+      : ({ label: item, value: item })));
+};
+
+/** Function to return the filter options based on the selected action value. */
+export const subActionOptions = (value, filterOptions) => {
+  switch (value) {
+    case 'vm':
+      return filterOptions.vm_options;
+    case 'miq_template':
+      return filterOptions.template_options;
+    case 'host':
+    case 'host_check_compliance':
+      return filterOptions.host_options;
+    case 'container_image':
+    case 'container_image_check_compliance':
+      return filterOptions.container_image_options;
+    case 'emscluster':
+      return filterOptions.cluster_options;
+    case 'storage':
+      return filterOptions.storage_options;
+    case 'vm_check_compliance':
+      return filterOptions.vm_options;
+    default:
+      return filterOptions.vm_options;
```

Comment:
That's a lot of case statements.  Maybe not for here, but it would be great to simplify this so `options` just gets the data from the right place.

Reply:
@jrafanie Made the changes.